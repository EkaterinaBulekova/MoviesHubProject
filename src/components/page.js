import React from "react";

export class Page extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
    };
  }

  render() {
    return (
      <div className="Page">
            <h1>{this.props.name}</h1>
      </div>
    );
  }
}
