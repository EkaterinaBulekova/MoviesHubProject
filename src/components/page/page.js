import React from "react";
import { Header } from "../header/header";
import MovieDetail from "../movie-detail/movie-detail";
import Button from "../button/button";
import Search from "../search/search";
import { MoviesContainer } from "../movies-container/movies-container";
import { ErrorBoundary } from "../error-boundary/error-boundary";
import FilmsByGenre from '../films-by-genre/films-by-genre';
import FilmsResults from '../films-results/films-results';
import getData from '../../utils/data-provider';

export class Page extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      searchBy: "title",
      search: "",
      sortBy: "release_date",
      selectedMovie: null,
      count: 0
    };
  }

  componentDidMount(){
    if (process.env.NODE_ENV =='development'){
      console.log('Page is created');
    }
  }

  initState(){
    this.setState({
      searchBy: "title",
      search: "",
      sortBy: "release_date",
      selectedMovie: null,
      count: 0
    })
  }

  handleSearchClick(data){
    if (data.search){
      this.setState({
        searchBy: data.searchBy,
        search: data.search
      });
    } 
    else
      this.initState();
  }

  handleMovieClick(movie){
    this.setState({
      searchBy: "genres",
      search: movie.genres,
      selectedMovie: movie,
    });
  }

  handleSortByClick(sortBy){
    this.setState({
      sortBy: sortBy
    });
  }

  render() {
    let selectedMovie = this.state.selectedMovie;
    let filter = {search: this.state.search, searchBy: this.state.searchBy, sortBy: this.state.sortBy}

    return (
      <div className="page">
        {<ErrorBoundary className="global-error">
          {(selectedMovie)
          ? <Header name={this.props.name}>
              <Button className="search-button" name="SEARCH" onClick={() => this.initState()}/>
              <MovieDetail movie={selectedMovie}/>
              <FilmsByGenre genres={selectedMovie.genres}/>
            </Header>
          : <Header name={this.props.name}>
              <Search onSearchClick={(value) => this.handleSearchClick(value)}/>
              <FilmsResults count={this.state.count} sortBy={this.state.sortBy} onClick={(value)=>this.handleSortByClick(value)}/>
            </Header>}
        </ErrorBoundary>}
        <ErrorBoundary className="global-error">
          <MoviesContainer filter={filter} getData={getData} onClick={(movie) => this.handleMovieClick(movie)} onUpdate={(count)=>this.setState({count: count})}></MoviesContainer>
        </ErrorBoundary>
      </div>
    );
  }
}
