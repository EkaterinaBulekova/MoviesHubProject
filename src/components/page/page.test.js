import React from 'react';
import { mount } from 'enzyme';
import { Page } from './page';
import toJson from 'enzyme-to-json';

describe('<Page />', () => {
  it('should render as expect', () => {
    const component = mount(<Page name={'test-name'}/>);
    component.instance().handleMovieClick = jest.fn();
    component.instance().handleSearchClick = jest.fn();
    component.instance().handleSortByClick = jest.fn();
    component.update();

    const searchButton = component.find('button.search-button');

    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('div.header-name').text()).toBe('test-name');
    searchButton.simulate('click');
    expect(component.instance().handleSearchClick).toHaveBeenCalledTimes(1);

    component.unmount();
  });
});