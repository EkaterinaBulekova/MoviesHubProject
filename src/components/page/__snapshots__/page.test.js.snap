// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render as expect 1`] = `
<Page
  name="test-name"
>
  <div
    className="page"
  >
    <ErrorBoundary
      className="global-error"
    >
      <Header
        name="test-name"
      >
        <div
          className="app-header"
        >
          <div
            className="header-name"
          >
            test-name
          </div>
          <Search
            onSearchClick={[Function]}
          >
            <div
              className="search-component"
            >
              <CustomInput
                className="search-field"
                onChange={[Function]}
                onEnter={[Function]}
                title="FIND YOUR MOVIE"
              >
                <div
                  className="search-field"
                >
                  <div
                    className="search-field-title"
                  >
                    FIND YOUR MOVIE
                  </div>
                  <input
                    className="search-field-input"
                    onChange={[Function]}
                    onKeyUp={[Function]}
                  />
                </div>
              </CustomInput>
              <SearchFilter
                onClick={[Function]}
                searchBy="title"
              >
                <ButtonsGroup
                  active="title"
                  group={
                    Object {
                      "buttons": Array [
                        Object {
                          "name": "TITLE",
                          "result": "title",
                        },
                        Object {
                          "name": "GENRE",
                          "result": "genres",
                        },
                      ],
                      "className": "search-filter",
                      "title": "SEARCH BY",
                    }
                  }
                  onClick={[Function]}
                >
                  <div
                    className="search-filter"
                  >
                    <div
                      className="search-filter-title"
                    >
                      SEARCH BY
                    </div>
                    <Button
                      className="search-filter-button active"
                      key="title"
                      name="TITLE"
                      onClick={[Function]}
                    >
                      <button
                        className="btn search-filter-button active"
                        name="TITLE"
                        onClick={[Function]}
                      >
                        TITLE
                      </button>
                    </Button>
                    <Button
                      className="search-filter-button"
                      key="genres"
                      name="GENRE"
                      onClick={[Function]}
                    >
                      <button
                        className="btn search-filter-button"
                        name="GENRE"
                        onClick={[Function]}
                      >
                        GENRE
                      </button>
                    </Button>
                  </div>
                </ButtonsGroup>
              </SearchFilter>
              <Button
                className="search-button active"
                name="SEARCH"
                onClick={[Function]}
              >
                <button
                  className="btn search-button active"
                  name="SEARCH"
                  onClick={[Function]}
                >
                  SEARCH
                </button>
              </Button>
            </div>
          </Search>
          <FilmsResults
            count={0}
            onClick={[Function]}
            sortBy="release_date"
          >
            <div
              className="films-results"
            />
          </FilmsResults>
        </div>
      </Header>
    </ErrorBoundary>
    <ErrorBoundary
      className="global-error"
    >
      <MoviesContainer
        filter={
          Object {
            "search": "",
            "searchBy": "title",
            "sortBy": "release_date",
          }
        }
        getData={[Function]}
        onClick={[Function]}
        onUpdate={[Function]}
      />
    </ErrorBoundary>
  </div>
</Page>
`;
